### DISCUSSION TITLE
Code Help - Python version of dovi_tool

### DISCUSSION THREAD

**Discussion by MattGeale** [quietvoid/dovi_tool#296] [General]
Hi all,

I'm not trying to create the exact same tool, but rather integrate the logic of the dovi_tool into a script that I'm using for my media. Said script remuxes a file and strips out elements I dont want based on filters I've setup... I'm wanting to do an analysis on the remuxed file after the script has reached that stage and I do like to organise my media in such a way that it is in the various video formats - the OCD part in me likes it :P

Anyway, at some level, I'm guessing dovi_tool uses ffprobe given its wide array of analytics so I'm hoping someone could point me in the right direction. I've got this in my python code which does generate a fair amount of data, however, I can't seem to come up with something that can accurately parse the JSON data because every file structures DOVI and HDR10+ differently - at least what ffprobe is showing me.

        `command = [
            self.ffprobe_path,
            "-hide_banner",
            "-loglevel", "quiet",
            "-print_format", "json",
            "-select_streams", "v",
            "-show_streams",
            "-show_entries", "stream=index,codec_name,codec_type,width,height,color_space,color_primaries,color_transfer,side_data_list",
            "-i", remuxed_file
        ]`

Whenver I run a file into ffprobe using those commands, sometimes its the same structure in side_data_list, and sometimes its just incomprehensibly different than what I'm expecting.

I've played around with the dovi_tool and it does seem to be very, very robust and accurate at spitting out data so I'm thinking my command to generate the JSON is wrong so I'm hoping I could be steered in the right direction.

Any help would be appreciated, I'm sure I'm going to be answering a bunch of questions :)

Thanks in advance

**Comment by quietvoid** [comment]
`dovi_tool` does not use ffprobe at all..
I don't understand what you're trying to do.

**Reply by MattGeale** [reply]
Oh lmao. In that case, carry on :P 
I assumed it used FFProbe - I'll delete the thread :)

**Reply by danka12324** [reply]
Hi there! It sounds like you're trying to analyze video files and extract information about their properties, such as codec, resolution, and color space. Your approach of using ffprobe to generate JSON data is a good one, and it's likely that the differences you're seeing in the structure of the JSON data are due to the different video formats and codecs used in the files you're analyzing.

To help you parse the JSON data, you might want to consider using a Python library like json or jsonpath-ng to extract the information you need. These libraries can help you navigate the JSON structure and extract specific values, even if the structure varies between files.

As for the dovi_tool, it's a great tool for analyzing video files, but it's not always straightforward to integrate its logic into your own script. You might want to consider looking into the source code of dovi_tool to see how it parses the JSON data and extracts the information you need. This might give you some ideas on to improve your own script.

I hope this helps! If you have any more questions or need more specific guidance, feel free to ask.
